src/schema/v2/artworksForUser/helpers.ts

```diff
@@ -1,21 +1,46 @@
 import { ResolverContext } from "types/graphql"
 import { CursorPageable } from "relay-cursor-paging"
-import { extractNodes } from "lib/helpers"
+import { convertConnectionArgsToGravityArgs, extractNodes } from "lib/helpers"
 import gql from "lib/gql"
 
-export const getNewForYouRecs = async (
+export const getNewForYouArtworkIDs = async (
   args: CursorPageable,
   context: ResolverContext
 ): Promise<string[]> => {
   const {
     authenticatedLoaders: {
       vortexGraphqlLoader: vortexGraphQLAuthenticatedLoader,
+      auctionLotRecommendationsLoader: auctionLotRecommendationsAuthenticatedLoader,
     },
     unauthenticatedLoaders: {
       vortexGraphqlLoader: vortexGraphQLUnauthenticatedLoader,
+      auctionLotRecommendationsLoader: auctionLotRecommendationsUnauthenticatedLoader,
     },
     xImpersonateUserID,
   } = context
+
+  if (args.includeAuctionArtworks) {
+    if (!auctionLotRecommendationsAuthenticatedLoader)
```

Comment:
Even in the case of an app request via impersonation, this loader will exist, so we can check it (even tho for impersonate queries we eventually use the unauthenticated loader).